build-logic/documentation/src/main/groovy/gradlebuild/docs/GradleBuildDocumentationPlugin.java

```diff
@@ -131,9 +130,8 @@ private void addUtilityTasks(TaskContainer tasks, GradleDocumentationExtension e
                 task.getProject().getExtensions().getByType(JavaToolchainService.class)
                     .launcherFor(spec -> spec.getLanguageVersion().set(JavaLanguageVersion.of(21)))
             );
-            task.workingDir(extension.getDocumentationRenderedRoot());
-            task.getMainModule().set("jdk.httpserver");
-            task.args("-p", String.valueOf(webserverPort));
+            task.getDocsDirectory().convention(extension.getDocumentationRenderedRoot());
+            task.getPort().convention(webserverPort);
```

Comment:
❓ Why conventions instead of set here?  We're creating the specific task for the docs server, is there a reason to have a default that could be overridden rather than just setting the value?

Reply:
This is just a habit to use convention in a plugin and set in build scripts.

It's not necessary, but it would make sure that if you did:
```
tasks.withType(ServeDocs) {
    port = 1234
}
```
in a build script, it would work as expected.